### PR TITLE
Add overload to deserialize headers case insensitively

### PR DESCRIPTION
**Pull request by jianghaolu**

### CODE REVIEW

**Comment by sima-zhu** [comment]
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=143683&view=logs&s=96ac2280-8cb4-5df5-99de-dd2da759617d

File: sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/jackson/JacksonAdapter.java [hunk 1]
```diff
@@ -70,7 +81,9 @@ public JacksonAdapter() {
                 .registerModule(AdditionalPropertiesSerializer.getModule(flatteningMapper))
                 .registerModule(AdditionalPropertiesDeserializer.getModule(flatteningMapper))
                 .registerModule(FlatteningSerializer.getModule(simpleMapper()))
-                .registerModule(FlatteningDeserializer.getModule(simpleMapper()));    }
+                .registerModule(FlatteningDeserializer.getModule(simpleMapper()));
+        headerMapper = initializeHeaderMapper(new ObjectMapper());
```
Comment (anuchandy) [inline sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/jackson/JacksonAdapter.java]: jfyi we could also do: 

```
headerMapper  = simpleMapper
    .copy()
    .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);
```
https://static.javadoc.io/com.fasterxml.jackson.core/jackson-databind/2.9.8/com/fasterxml/jackson/databind/ObjectMapper.html#copy--

**Comment by sima-zhu** [comment]
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=143743&view=logs&s=96ac2280-8cb4-5df5-99de-dd2da759617d

**Pull request merged**